### ISSUE TITLE
Feature: Add skip button when deleting multiple files that are in use by another application

### ISSUE THREAD

**Issue by AzaelMew** [files-community/Files#8239]
### Solution/Idea

When deleting a lot of items from for example Temp folder, a lot of items are usually in use, would be convenient if there was a way to say "ok" to every single file in there, so there's no need to say "ok" every time

Also would be nice if when deleting items, there was a way to see what process is using it

![image](https://user-images.githubusercontent.com/66936848/153030086-c92266e3-c127-4d4f-a190-fe9516c9cebd.png)

### Alternatives

na

### Priorities

| Capability | Priority |
- Replace "okay" with a skip button
- Add skip all if there are multiple files in use

**Comment by BharatBaweja** [comment]
@yaira2 Hi
is this  issue is still open? I am interested in working on this

**Comment by BharatBaweja** [comment]
Ok, I will start working on it

**Comment by yaira2** [comment]
@BharatBaweja let us know if you have any questions.

**Comment by William-Hiatt** [comment]
@yaira2 it looks like having a pop up with what program is using the file is already implemented. The skip button hasn't been implemented yet, do you think the skip button is still a good first issue?

**Comment by yaira2** [comment]
I think so, do you have any feedback on this or alternative ideas?

**Comment by sirwilliamthefirst** [comment]
I am interested in working on this issue, but if I open lets say an mp3 file, then delete it depsite it being used by another program I actually dont get any dialogue, and it just deletes

**Comment by 0x5bfa** [comment]
Yeah if the program that opened the mp3 file allows other programs delete, other apps can delete. There's an option called "shared mode" when opening a file.

If we are to implement this, we should make use of IShellOperation's skip, which FE also uses.

**Comment by yaira2** [comment]
> I am interested in working on this issue, but if I open lets say an mp3 file, then delete it depsite it being used by another program I actually dont get any dialogue, and it just deletes

You might need to try with a different file type (eg exe).

**Comment by workbysaran** [comment]
> If we are to implement this, we should make use of IShellOperation's skip, which FE also uses.

@hishitetsu 
After reviewing the existing codebase, I have found that it currently uses a post-operation dialog method to interact with the user for all file operations (Copy, Move, Delete) and the existing pre-operation callback is used only for the progress tracking.

Should I implement the Skip button while keeping the existing post-operation dialog pattern or should I refactor the entire codebase to use pre-operation callbacks for interactive skip functionality?

**Comment by mustafadawoodi** [comment]
Is this issue still open?

**Comment by Josh65-2201** [comment]
Yes

**Comment by codenano077** [comment]
I'd like to work on this! I'll look into the IShellOperation skip logic and implement the Skip and Skip All buttons for the delete dialog.

**Comment by yair100** [comment]
@AzaelMew this issue will be resolved in the next release.
